lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4872,14 +4838,34 @@ ByteCodeGenerator::GetLdSlotOp(Scope *scope, int envIndex, Js::RegSlot scopeLoca
     return op;
 }
 
-void ByteCodeGenerator::EmitPropLoadThis(Js::RegSlot lhsLocation, ParseNode *pnode, FuncInfo *funcInfo, bool chkUndecl)
+bool ByteCodeGenerator::ShouldLoadConstThis(FuncInfo* funcInfo)
 {
-    this->EmitPropLoad(lhsLocation, pnode->sxPid.sym, pnode->sxPid.pid, funcInfo, true);
+    // Load a const 'this' binding if the following holds
+    // - The function has a 'this' name node
+    // - We are in a global or global lambda function
+    // - The function has no 'this' symbol (an indirect eval would have this symbol)
+    return funcInfo->thisConstantRegister != Js::Constants::NoRegister
```

Comment:
Instead of checking that we're in a global context (global func or global lambda), can we assert it? I.e., if "this" has no symbol, we should be in a global "this" context.

Reply:
Isn't it possible for an ordinary function to not have a 'this' binding? I guess we could just check to see if `funcInfo->thisConstantRegister != Js::Constants::NoRegister` and then assert and skip this check function.

Reply:
That sounds right.